tabs/firmware_flasher.js

```diff
@@ -329,11 +333,14 @@ TABS.firmware_flasher.initialize = function (callback) {
             function failed_to_load() {
                 $('span.progressLabel').text(chrome.i18n.getMessage('firmwareFlasherFailedToLoadOnlineFirmware'));
                 $('a.flash_firmware').addClass('disabled');
+                enable_load_online_button();
             }
 
             var summary = $('select[name="firmware_version"] option:selected').data('summary');
             if (summary) { // undefined while list is loading or while running offline
+                $("a.load_remote_file").text(chrome.i18n.getMessage('firmwareFlasherButtonLoading')).addClass('disabled');
```

Comment:
if we have separate function to enable, why not for disabling?

Reply:
There are 2 disabled states. One when there's no firmware selected in the top comboboxes and another one while the firmware is being downloaded. The button will show different texts. There's only one enabled state, so it extracted that to a function.

Reply:
OK, understand